python/health-check/src/health_check/exporters/supportconfig_exporter.py

```diff
@@ -0,0 +1,583 @@
+# SPDX-FileCopyrightText: 2023 SUSE LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+"""
+Main supportconfig exporter module that collects metrics into a json file
+and serves the file by using an HTTP server.
+"""
+
+from collections import defaultdict, namedtuple
+import os
+from pathlib import Path
+import re
+import signal
+import sys
+from typing import Dict, Tuple
+import yaml
+import json
+
+import http.server
+
+import static_metrics
+from static_metrics import metrics_config
+
+
+def sigterm_handler(**kwargs):
+    del kwargs  # unused
+    print("Detected SIGTERM. Exiting.")
+    sys.exit(0)
+
+
+signal.signal(signal.SIGTERM, sigterm_handler)
+
+
+class SupportConfigMetricsCollector:
+    """A collector that collects metrics, and exports them into a file."""
+
+    def __init__(self, supportconfig_path=None):
```

Comment:
We require the path, then why do we set it to None...

```suggestion
    def __init__(self, supportconfig_path: str):
```